lib/ethpy/ethpy/hyperdrive/transactions.py

```diff
@@ -55,14 +49,13 @@ def get_hyperdrive_pool_info(hyperdrive_contract: Contract, block_number: BlockN
     dict[str, Any]
         A dictionary containing the Hyperdrive pool info returned from the smart contract.
     """
-    return hypertypes_pool_info_to_fixedpoint(
-        contract_pool_info_to_hypertypes(
-            smart_contract_read(hyperdrive_contract, "getPoolInfo", block_number=block_number)
-        )
-    )
+    pool_info = hyperdrive_contract.functions.getPoolInfo().call(None, block_number)
```

Comment:
🤩

Reply:
BTW, this looks very much like what web3 already exposes:
https://web3py.readthedocs.io/en/stable/web3.contract.html#web3.contract.ContractFunction
In other words, I'm pretty sure before this change you can do this same syntax on the underlying `Contract` object